drivers/display/ft8xx/ft8xx.c

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright (c) 2020 Hubert Miś
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <drivers/display/ft8xx/ft8xx.h>
+
+#include <stddef.h>
+#include <stdint.h>
+
+#include <device.h>
+#include <kernel.h>
+#include <logging/log.h>
+
+#include <drivers/display/ft8xx/ft8xx_copro.h>
+#include <drivers/display/ft8xx/ft8xx_dl.h>
+
+#include "ft8xx_common.h"
+#include "ft8xx_drv.h"
+#include "ft8xx_host_commands.h"
+#include "ft8xx_memory.h"
+
+#define DT_DRV_COMPAT ftdi_ft800
+
+#define LOG_MODULE_NAME ft8xx
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+#define PCLK      DT_INST_PROP(0, pclk)
+#define PCLK_POL  DT_INST_PROP(0, pclk_pol)
+#define CSPREAD   DT_INST_PROP(0, cspread)
+#define SWIZZLE   DT_INST_PROP(0, swizzle)
+#define VSIZE     DT_INST_PROP(0, vsize)
+#define VOFFSET   DT_INST_PROP(0, voffset)
+#define VCYCLE    DT_INST_PROP(0, vcycle)
+#define VSYNC0    DT_INST_PROP(0, vsync0)
+#define VSYNC1    DT_INST_PROP(0, vsync1)
+#define HSIZE     DT_INST_PROP(0, hsize)
+#define HOFFSET   DT_INST_PROP(0, hoffset)
+#define HCYCLE    DT_INST_PROP(0, hcycle)
+#define HSYNC0    DT_INST_PROP(0, hsync0)
+#define HSYNC1    DT_INST_PROP(0, hsync1)
+
+#define DLSWAP_FRAME 0x02
+
+#define FT8XX_EXPECTED_ID 0x7C
+
+static int_callback irq_callback;
+
+static void command(uint8_t cmd)
+{
+	(void)ft8xx_drv_command(cmd);
```

Comment:
error chain lost here, better propagate errors

Reply:
Unfortunately FT800 API defines functions as `void`. Because of that it's hard to propagate errors. I've decided to just log errors to keep compatibility with API suggested in FT800 documentation.

Reply:
`ft8xx_drv_command` is defined in this file and returns an error code if something goes wrong, so it should be propagated accordingly

Reply:
Please have a look at this snippet from FT800 Programmers Guide. 
![image](https://user-images.githubusercontent.com/21953319/103685789-161ece00-4f8e-11eb-9589-fd072f27bfdb.png)

Signature this function is expected to be `void host_command(uint8_t command)`. I'm not sure what action should be taken here if `ft8xx_drv_command()` returns an error. Assertion fault?

Reply:
I've seen many examples/guides that do not take into account error handling, but real applications need to. If not, low level failures get buried resulting in hard to debug problems. As implementation is not external here, and you have the choice of adding error handling, I think it should be added.

Reply:
> you have the choice of adding error handling, I think it should be added

I understand your concern. However, creating own API for this device instead of following FT800 Programmers Guide would make hard porting applications using this display from other platforms. If the driver follows defined API, I can simply reuse display software modules from other projects. In similar way, I can reuse network modules using BSD socket API, because Zephyr follows defined socket API, instead of own improved implementation.

I think it would be better to decide on the best way of correct handling of low level errors inside the driver and keep API compatibility with the specification. My current proposal is to use error logs. I can implement something else, like a callback function called when operation fails.

What do you think?

Reply:
In such case, then I'd place an assertion inside these functions so that when enabled the point of failure will be known.